common/types/transaction.go

```diff
@@ -0,0 +1,127 @@
+package types
+
+import (
+	"github.com/spacemeshos/ed25519"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/signing"
+	"github.com/spacemeshos/sha256-simd"
+)
+
+type TransactionId Hash32
+
+func (id TransactionId) Hash32() Hash32 {
```

Comment:
Most `TransactionId`s are not pointers, since `Transaction.Id()` returns a non-pointer. Having pointer receiver methods requires a bunch of boilerplate and code changes that I didn't think was justified. Let's discuss in person if you think this is important.